src/riak_core_ring_manager.erl

```diff
@@ -721,5 +732,14 @@ do_write_ringfile_test() ->
     ?assertMatch({error,_}, do_write_ringfile(GenR(ring_perms), ?TEST_RINGFILE)),
     ok = file:change_mode(?TEST_RINGDIR, 8#00755).
 
--endif.
+is_stable_ring_test() ->
+    % This test is efferently coupled to the PROMOTE_TIMEOUT macro value
+    {A,B,C} = Now = os:timestamp(),
+    Within = {A,B-45,C},
```

Comment:
This test will break if `?PROMOTE_TIMEOUT` is changed. Considering it's unlikely for the unit to be changed from ms, since that would require changing other code, I'd change this test so that it calculates the values in terms of `?PROMOTE_TIMEOUT` Something like:
```erlang
TimeoutSecs = ?PROMOTE_TIMEOUT div 1000,
Within = {A, B - (TimeoutSecs div 2), C},
Outside = {A, B - (TimeoutSecs + 1), C}
```

Technically this still has a chance of failing every 277 days if B is < ?TIMEOUT_SECS  or ?TIMEOUT_SECS/2, but I'm not really worried about that.